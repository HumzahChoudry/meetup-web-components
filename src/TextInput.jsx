import React from 'react';
import cx from 'classnames';

/**
 * @module TextInput
 */
class TextInput extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
			value: props.value || '',
		};
		this.onChange = this.onChange.bind(this);
	}

	onChange(e) {
		this.setState({ value: e.target.value });
	}

	render() {
		const {
			name,
			value,	// eslint-disable-line no-unused-vars
			label,
			labelClassName,
			className,
			children,
			error,
			required,
			...other
		} = this.props;

		const classNames = cx(
			{ 'field--error': error },
			className
		);

		const labelClassNames = cx(
			{ required },
			labelClassName
		);

		return (
			<div>
				<label className={labelClassNames} htmlFor={other.id}>
					{label}
				</label>

				<input type='text'
					name={name}
					value={this.state.value}
					required={required}
					className={classNames}
					onChange={this.onChange}
					{...other} />

				{ this.props.maxLength && <p className='text--caption align--right'>{this.state.value.length} / {this.props.maxLength}</p> }

				{ error && <p className='text--error'>{error}</p> }
				{children}
			</div>
		);
	}
}

TextInput.propTypes = {
	name: React.PropTypes.string.isRequired,
	error: React.PropTypes.string,
	label: React.PropTypes.string,
	labelClassName: React.PropTypes.string,
	required: React.PropTypes.bool
};

export default TextInput;
