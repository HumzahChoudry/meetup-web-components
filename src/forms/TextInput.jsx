import React from 'react';
import cx from 'classnames';

/**
 * @module TextInput
 */
class TextInput extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
			value: props.value || '',
		};
		this.onChange = this.onChange.bind(this);
	}

	getInputType() {
		return this.props.isSearch ? 'search' : 'text';
	}

	onChange(e) {
		this.setState({ value: e.target.value });
		if (this.props.onChange) {
			this.props.onChange(e);
		}
	}

	render() {
		const {
			name,
			value, // eslint-disable-line no-unused-vars
			label,
			labelClassName,
			className,
			children,
			error,
			required,
			id,
			onChange, // eslint-disable-line no-unused-vars,
			maxLength,
			pattern
		} = this.props;

		const classNames = cx(
			{ 'field--error': error },
			className
		);

		const labelClassNames = cx(
			{ required },
			labelClassName
		);

		return (
			<div>
				<label className={labelClassNames} htmlFor={id}>
					{label}
				</label>

				<input type={this.getInputType()}
					name={name}
					value={this.state.value}
					required={required}
					className={classNames}
					onChange={this.onChange}
					maxLength={maxLength}
					pattern={pattern} />

				{ this.props.maxLength && <p className='text--caption align--right'>{this.state.value.length} / {this.props.maxLength}</p> }

				{ error && <p className='text--error'>{error}</p> }
				{children}
			</div>
		);
	}
}

TextInput.propTypes = {
	name: React.PropTypes.string.isRequired,
	error: React.PropTypes.oneOfType([
		React.PropTypes.string,
		React.PropTypes.element
	]),
	id: React.PropTypes.string,
	maxLength: React.PropTypes.number,
	pattern: React.PropTypes.string,
	label: React.PropTypes.oneOfType([
		React.PropTypes.string,
		React.PropTypes.element
	]),
	labelClassName: React.PropTypes.string,
	required: React.PropTypes.bool,
	isSearch: React.PropTypes.bool,
	onChange: React.PropTypes.func,
};

export default TextInput;
