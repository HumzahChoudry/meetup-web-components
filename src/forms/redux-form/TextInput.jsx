import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../TextInput';

/**
 * @param {Object} props - props passed in from parent and redux-form
 * @description wraps standard TextInput web component for use with redux-form
 * deconstructs props that redux-forms sets and sets them on TextInput
 * @return {Component} TextInput
 */
const ReduxFormTextInput = props => {
	const { meta, input, validateBeforeTouched, ...other } = props;
	const error = (validateBeforeTouched || meta.touched || meta.submitFailed)
		? meta.error
		: null;

	return <TextInput error={error} {...input} {...other} />;
};

ReduxFormTextInput.propTypes = {
	meta: PropTypes.object.isRequired,
	input: PropTypes.element.isRequired,
	validateBeforeTouched: PropTypes.bool,
};

ReduxFormTextInput.displayName = 'ReduxFormTextInput';

export default ReduxFormTextInput;
