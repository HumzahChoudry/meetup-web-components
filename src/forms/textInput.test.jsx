import React from 'react';
import TextInput from './TextInput';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('TextInput', function() {
	const onChange = jest.fn();
	const LABEL_TEXT = 'Super Hero',
		VALUE = 'Batman',
		NAME_ATTR = 'superhero',
		MAX_LEN = 20,
		ERROR_TEXT = 'Too wimpy.';

	let textInputComponent,
		inputEl;

	const formAttrs = {
		id: NAME_ATTR,
		maxLength: MAX_LEN,
		required: true,
		error: ERROR_TEXT,
	};

	beforeEach(() => {
		textInputComponent = mount(
			<TextInput
				name={NAME_ATTR}
				label={LABEL_TEXT}
				value={VALUE}
				onChange={onChange}
				{...formAttrs}
			/>
		);

		inputEl = textInputComponent.find('input');
	});

	afterEach(() => {
		textInputComponent = null;
		inputEl = null;
	});

	it('matches snapshot', () => {
		expect(toJson(textInputComponent)).toMatchSnapshot();
	});

	it('should have a name prop', () => {
		expect(inputEl.prop('name')).toEqual(NAME_ATTR);
	});

	it('should have a value prop when one is specified', () => {
		expect(inputEl.prop('value')).toEqual(VALUE);
	});

	it('should have a label when label is given', () => {
		expect(() => textInputComponent.find('label')).not.toBeNull();
		const labelEl = textInputComponent.find('label').getDOMNode();
		expect(labelEl.textContent).toEqual(LABEL_TEXT);
	});

	it('should have prop type search if `isSearch` is set to true', () => {
		textInputComponent = mount(
			<TextInput
				name={NAME_ATTR}
				label={LABEL_TEXT}
				defaultValue={VALUE}
				isSearch
			/>
		);

		inputEl = textInputComponent.find('input');
		expect(inputEl.prop('type')).toEqual('search');

	});

	it('should have a disabled prop when specified', () => {
		textInputComponent = mount(
			<TextInput
				name={NAME_ATTR}
				label={LABEL_TEXT}
				defaultValue={VALUE}
				disabled
			/>
		);

		inputEl = textInputComponent.find('input');
		expect(inputEl.prop('disabled')).not.toBeNull();
	});

	it('should have a required prop when specified', () => {
		expect(inputEl.prop('required')).not.toBeNull();
	});

	it('should specify attribute props that are passed in', function() {
		expect(inputEl.prop('maxLength')).toEqual(MAX_LEN);
	});

	it('should have an error when one is specified', function() {
		const errorEl = textInputComponent.find('.text--error').getDOMNode();
		expect(errorEl.textContent).toEqual(ERROR_TEXT);
	});

	it('should call onChange `props` function when input is changed', () => {
		const newValue = `${VALUE}r`;
		const boundComponent = mount(
			<TextInput
				name={NAME_ATTR}
				label={LABEL_TEXT}
				value={VALUE}
				onChange={onChange}
			/>
		);
		inputEl = boundComponent.find('input');
		inputEl.simulate('change', { target: { value: newValue }});

		expect(onChange).toHaveBeenCalled();
	});
});
