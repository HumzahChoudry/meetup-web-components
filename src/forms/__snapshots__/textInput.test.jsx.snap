// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders a required HTML <input> with expected attributes for mock data 1`] = `
<div>
  <div
    className="inputContainer"
  >
    <label
      className="label--field required"
      htmlFor="superhero"
    >
      Super Hero
    </label>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <input
        className="span--100"
        id="superhero"
        name="superhero"
        required={true}
        type="text"
        value="Batman"
      />
    </div>
  </div>
</div>
`;
